packages/core/src/types/index.ts

```diff
@@ -67,12 +68,18 @@ export function normalizeTypes(
     } as any;
   }
 
-  for (const type of Object.values(normalizedTypes)) {
-    normalizeType(type);
+  if (incorrectRefs) {
+    for (const type of Object.values(normalizedTypes)) { normalizeIncorrectRefs(type, incorrectRefs); }
   }
-
+  for (const type of Object.values(normalizedTypes)) { normalizeType(type); }
```

Comment:
keep the original loop and just add normalizeIncorrectRefs call inside it.
No need for two loops.